modules/go/main.go

```diff
@@ -142,6 +152,82 @@ type Go struct {
 
 	// Enable go experiments
 	Experiment []string
+
+	Include []string
+
+	Exclude []string
+}
+
+func (p Go) GenerateDaggerRuntimes(ctx context.Context) (Go, error) {
+	mods, err := p.Modules(ctx)
+	if err != nil {
+		return p, err
+	}
+	daggerMods := map[string]*dagger.Directory{}
```

Comment:
It seems that the goroutines spawned by `parallel.New().WithLimit(3)` concurrently assign into `daggerMods`, which is not fully concurrent safe, example: https://go.dev/play/p/JX-jo2sPsQI

It's only 3 goroutines, on a few different paths here, but, unless we can assure that `daggerModPath` is different in each goroutine, there's a risk (non blocking as it's very very unlikely)

Reply:
Concurrency is not an issue in this case, because all the goroutines write the same thing: `nil`. So even if several goroutines write to the same key, it doesn't matter which writer wins: they're all writing the same value.

Reply:
Oh!!! Well we don't want that! OK I'll add a mutex. Thanks!